bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/BigAssFanHandlerFactory.java

```diff
@@ -53,7 +53,10 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
     protected @Nullable ThingHandler createHandler(Thing thing) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
         if (SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID)) {
-            return new BigAssFanHandler(thing, networkAddressService.getPrimaryIpv4HostAddress());
+            String address = networkAddressService.getPrimaryIpv4HostAddress();
+            if (address != null) {
+                return new BigAssFanHandler(thing, address);
+            }
```

Comment:
I'm not sure if good or bad, but there is a functional change here. Previously, if returned address was null, handler would still be created. This would then create a new `ConnectionManager` with address being null. Then [InetAddress.getByName](https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#getByName-java.lang.String-) would be called with null argument which would return a representation of the loopback interface. Just something to consider. At least there should be a warning.

You might also want to check other implementations - for example squeezebox and chromecast.

Reply:
I think it is needed here. As checking for null (even on a local var) and if null scheduling a task might cause multiple task to be scheduled. For the other way around (safely accessing a property) i agree.
When looking for `synchronized void start` i found many bindings scheduling a task like this. There are also more bindings that have the `synchronised` block on this within the method, so there might be many more bindings.

Reply:
Changed it and made it similar to many other bindings. When `getPrimaryIpv4HostAddress` returns null a `warn` message is logged and the `createHandler` method is returned immediate.

Reply:
Can you mention some examples, so I could check? It's different from squeezebox and chromecast which I mentioned.

Reply:
You're right. The bindings i looked at all have this like chomecast. They wrapped the call to `getPrimaryIpv4HostAddress` in a `callbackurl` method and pass that to the handler constructor. In this case there is no wrapper, i'll come up with a fix. Thanks.

Reply:
Looked at it again, i changed it back to the original functionality. Both solutions have pros/cons. just lets get this forward